modules/rating/src/main/glicko2/RatingCalculator.scala

```diff
@@ -41,15 +49,17 @@ final class RatingCalculator(
 
   private val ratingPeriodsPerMilli: Double = ratingPeriodsPerDay.value * DAYS_PER_MILLI
 
-  /** <p>Run through all players within a resultset and calculate their new ratings.</p> <p>Players within the
-    * resultset who did not compete during the rating period will have see their deviation increase (in line
-    * with Prof Glickman's paper).</p> <p>Note that this method will clear the results held in the association
-    * resultset.</p>
+  /** Run through all players within a resultset and calculate their new ratings. Players within the resultset
+    * who did not compete during the rating period will have see their deviation increase (in line with Prof
+    * Glickman's paper). Note that this method will clear the results held in the association resultset.
     *
     * @param results
     */
-  def updateRatings(results: RatingPeriodResults[?], skipDeviationIncrease: Boolean = false) =
-    val players = results.getParticipants
+  def updateRatings(
+      players: Iterable[Rating],
+      results: RatingPeriodResults[?],
+      skipDeviationIncrease: Boolean = false
+  ) =
```

Comment:
this is strange, or I'm misunderstanding something.

This `updateRatings` function now takes an additional argument `players`.

That info used to come from `results` which also contains the players as `results.players`.

Which means there are now 2 sources of truth for the players. Both `players` and `results.players`.

I see that `results.players` has been made private and is completely unused (?!) but it is still a code smell in my opinion, and a potential source of bugs.

Reply:
Indeed, this PR exposes what in my mind was previously a code smell: `RatingCalculator.scala` looking at `results.players` to figure out "which ratings need to be updated?" was unnecessarily complex (besides that previous code being confusing in other ways with trying to figure out if a player played any games in that `RatingPeriodResults` or not).

Tutor could have exercises with fixed ratings: in fact, in chess literature (books, magazines, CDs, other training materials) exercises are provided with "if you get this score, we estimate your rating is X."  But this PR might be prematurely forcing us to make that design decision.

Reply:
For others' benefit; we discussed on stream that we don't think `results.players` needs to exist, but it's an artifact from a previous reference implementation prematurely optimized straight from java hell.

Reply:
Regarding `result.players`, class `Result` is perhaps the world's least efficient implementation of `Map[Rating, Score]`.  At present my head hurts trying to think about it.

Reply:
Oh wait... `Result` should actually be `Tuple[OpponentRating, Score]` and there's no need for any functions.